tiledb/sm/array_schema/domain.h

```diff
@@ -81,6 +81,13 @@ class Domain {
       const std::vector<shared_ptr<Dimension>> dimensions,
       Layout tile_order);
 
+  /** Constructor.
+   *
+   * @param dimensions The dimensions of the domain.
+   * @param dim_num The number of dimensions.
+   */
+  Domain(std::vector<shared_ptr<Dimension>> dimensions, unsigned dim_num);
```

Comment:
We don't want a new constructor. The one that already takes `vector<shared_ptr<Dimension>>` is sufficient. This new one isn't initializing all the member variables, notably cell and tile order.

It looks as if the code being modified here is not itself fully initializing the object.